prover/src/lib.rs

```diff
@@ -41,10 +40,9 @@ impl ProverInput {
         info!("Proving block {block_number}");
 
         let other_data = self.other_data;
-        let txs = self.block_trace.into_txn_proof_gen_ir(
-            &ProcessingMeta::new(resolve_code_hash_fn),
-            other_data.clone(),
-        )?;
+        let txs = self
+            .block_trace
+            .into_proof_gen_mpt_ir(&resolve_code_hash_fn, other_data.clone())?;
```

Comment:
I know this is still a draft, but I just want to give some extra info stemming from my comments from ([230](https://github.com/0xPolygonZero/zk_evm/pull/230)).

For now, I'm fine if we create an impl that has `resolve` as a todo or similar, just because currently all code mappings appear embedded in the pre-image.